bokehjs/src/lib/core/util/zoom.ts

```diff
@@ -10,6 +11,7 @@ type ScaleRanges = RangeInfo & {
 }
 
 export function scale_interval(range: Interval, factor: number, center?: number | null): Bounds {
+  assert(Math.abs(factor) < 1)
   const [min, max] = minmax(range.start, range.end)
```

Comment:
Maybe this would be simpler then the assert and a clamp with somewhat arbitrary cutoff values:
```suggestion
  const {start, end} = range
  if (factor >= 1) {
    return [start, end]
  }
  const [min, max] = minmax(start, end)
```
Then you also don't have to change the baseline.

Reply:
I guess I am missing something but this does not seem to be correct at all. If factor is greater-equal than one, we want to instead perform a scale with a factor near (but less than) one. But the change above is equivalent to setting factor equal to zero.

Reply:
Additionally, though I suppose clamping the negate side may not be required to avoid any bug, it does seem reasonable to apply the same clamp in both directions to maintain consistency.

Reply:
I agree 0.95 is somewhat arbitrary, but it just needs to be strictly less than one (`factor=1` has its own issue with letting range `start == end` which blows up for different reasons). Happy to change it to 0.99 or anything else suggested. Can include re-thinking this in the follow-on issue.